### DISCUSSION TITLE
Adding multi-site support

### DISCUSSION THREAD

**Discussion by nc7s** [11ty/eleventy#4052] [Enhancement Queue]
**Is your feature request related to a problem? Please describe.**

I have a number of small sites, which I'd prefer to put in one repo, because it's rather cumbersome to break them into their own repos, or run `yarn add -D @11ty/eleventy eleventy-plugin-example` in each site directory.

Relative: https://github.com/11ty/eleventy/issues/662, no formal outcome.

**Describe the solution you'd like**

It's possible to `yarn run eleventy --input=site-1/ --output=_sites/site-1/` for a given `site-1`, then for `site-2`, `site-3`, and so on.

But it's not possible to have a plugin that alters the build process for a single site to work on multiple sites.

So, several ways:

- Build the repo as if it's a single site, with each actual site being a directory, then publish only the directory for each site. This way all sites are built on each deployment of each site, and links must be taken care of.
- Specify the commands for each site in `package.json:scripts`, for example
  ```json
  {
    "scripts": {
      "build-site-1": "yarn run eleventy --input=site-1/ --output=_sites/site-1/",
      "build-site-2": "yarn run eleventy --input=site-2/ --output=_sites/site-2/",
      "build-site-3": "yarn run eleventy --input=site-3/ --output=_sites/site-3/",
    }
  }
  ```
  As you can see, it's verbose, repetitive, and error-prone.
- Have a rather independent builder plugin that adds a command `eleventy-multisite`, which reads some multi-site specific configuration in `.eleventy.js`, and runs `eleventy` accordingly, for each site.
  - This requires support for site-level configuration, possibly by merging site config into global for each site.
- Integrate into the main package. Adds bloat, so it's listed only as an option.

I'm experimenting with the second approach, but any suggestions and ideas are welcome.

**Comment by nc7s** [comment]
A prototype has been worked out: https://github.com/nc7s/eleventy-multisite, also on npm as package `eleventy-multisite`.

I've merely hand-tested a few times, needs more test cases. Otherwise, the features should be sufficient.

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!
